cmake/public/utils.cmake

```
@@ -537,11 +537,16 @@ endfunction()
 # CHeck if given flag is supported and append it to provided outputvar
 # Also define HAS_UPPER_CASE_FLAG_NAME variable
 # Usage:
-#   append_cxx_flag_if_supported("-Werror" CMAKE_CXX_FLAGS)
-function(append_cxx_flag_if_supported flag outputvar)
+#   append_torch_cxx_flag_if_supported("-Werror" CMAKE_CXX_FLAGS)
+function(append_torch_cxx_flag_if_supported flag outputvar)
     string(TOUPPER "HAS${flag}" _FLAG_NAME)
     string(REGEX REPLACE "[=-]" "_" _FLAG_NAME "${_FLAG_NAME}")
-    check_cxx_compiler_flag("${flag}" ${_FLAG_NAME})
+    if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```

are you sure that clang behaves the same way?
Also, please add a comment explaining why this is necessary 
```suggestion
    if(CMAKE_CXX_COMPILER_ID STREQUAL "GCC")
```

---

After a test I found it is GCC specific behavior.